homeassistant/components/omnilogic/sensor.py

```diff
@@ -0,0 +1,688 @@
+
+        self._name = (
+            self._backyard["BackyardName"]
+            + " "
+            + self.bow.get("Name")
+            + " "
+            + self.entitydata.get("Name")
+            + " Setting"
+        )
+
+        super().add_alarms(sensordata.get("Alarms"))
+
+        return self._state
+
+
+class OmniLogicPHSensor(OmnilogicSensor):
+    """Define an OmniLogic pH Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        device_class: str,
+        icon: str,
+        unit: str,
+        sensordata: dict,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            backyard=backyard,
+            bow=bow,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            sensordata=sensordata,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the pH sensor."""
+        _LOGGER.debug("Updating state of sensor: %s", self._name)
+
+        sensordata = super().find_csad(self.coordinator, self.entitydata["systemId"])
+
+        phstate = None
+        if sensordata.get("ph") != 0:
+            phstate = sensordata.get("ph")
+
+        self._state = phstate
+        self._unit = "pH"
+        self._name = self._backyard["BackyardName"] + " " + self.bow.get("Name") + " pH"
+
+        super().add_alarms(sensordata.get("Alarms"))
+
+        return self._state
+
+
+class OmniLogicORPSensor(OmnilogicSensor):
+    """Define an OmniLogic ORP Sensor."""
+
+    def __init__(
+        self,
+        coordinator: OmniLogicUpdateCoordinator,
+        kind: str,
+        name: str,
+        backyard: dict,
+        bow: dict,
+        device_class: str,
+        icon: str,
+        unit: str,
+        sensordata: dict,
+    ):
+        """Initialize the sensor."""
+        super().__init__(
+            coordinator=coordinator,
+            kind=kind,
+            name=name,
+            backyard=backyard,
+            bow=bow,
+            device_class=device_class,
+            icon=icon,
+            unit=unit,
+            sensordata=sensordata,
+        )
+
+    @property
+    def state(self):
+        """Return the state for the ORP sensor."""
+        _LOGGER.debug("Updating state of sensor: %s", self._name)
+
+        sensordata = super().find_csad(self.coordinator, self.entitydata["systemId"])
+
+        orpstate = None
+        if sensordata.get("orp") != -1:
+            orpstate = sensordata.get("orp")
+
+        self._state = orpstate
+        self._unit = "mV"
+        self._name = (
+            self._backyard["BackyardName"] + " " + self.bow.get("Name") + " ORP"
+        )
+
+        super().add_alarms(sensordata.get("Alarms"))
+
+        return self._state
+
+
+class OmniLogicAlarmSensor(OmnilogicSensor):
```

Comment:
This should be a binary sensor instead. Please remove it for now.

Reply:
Fair - agree. Have removed in next commit.